libkbfs/block_types.go

```diff
@@ -188,10 +188,30 @@ func (fb *FileBlock) NewEmpty() Block {
 
 // DataVersion returns data version for this block.
 func (fb *FileBlock) DataVersion() DataVer {
+	// If this is an indirect block, and none of its children are
+	// marked as direct blocks, then this must be a big file.  Note
+	// that we do it this way, rather than returning on the first
+	// non-direct block, to support appending to existing files and
```

Comment:
Hmm, I didn't think of adding it to `BlockInfo`.  But now that I think about it, I think the reason it makes sense there is because the list of block changes in the MD update contains `BlockPointer`s.  So including it in `BlockPointer` means that both CR, and other devices that get an update pushed to them, can infer something about the blocks.  That might be useful for future optimizations, though it isn't used at the moment.

I will add the panics you suggest.